orderer/consensus/etcdraft/chain_test.go

```diff
@@ -2385,11 +2385,20 @@ var _ = Describe("Chain", func() {
 					var newLeader, remainingFollower *chain
 					var c2state raft.SoftState
 					var c3state raft.SoftState
+					retry := 1
 					for newLeader == nil || remainingFollower == nil {
 						select {
 						case c2state = <-c2.observe:
 						case c3state = <-c3.observe:
 						case <-time.After(LongEventualTimeout):
+							// abdicateleader might fail to transfer the leadership when the next candidate
+							// busy with applying committed entries; in that case,
+							// send an artificial MsgTimeoutNow to node to pick next leader
+							if retry > 0 {
+								By("leadership transfer not complete, hence retrying")
+								c2.Consensus(&orderer.ConsensusRequest{Payload: protoutil.MarshalOrPanic(&raftpb.Message{Type: raftpb.MsgTimeoutNow, To: 2})}, 0)
```

Comment:
As per the design, if the leadership transfer is not completed then it is defined to complete in the next election timeout(ELECTION_TICK). Since we don't manipulate the election timeout, there are places in the unit text, we send out artificial MsgTimeOut to force the election(for only testing purpose).  It is documented/comment here: 
https://github.com/hyperledger/fabric/blob/2ef46bbc718de220bbd33ee4e1ce65194b963358/orderer/consensus/etcdraft/chain_test.go#L50

Reply:
OK and why are we sending to (2) and not to (3) also?

Reply:
we can send it to any node.  if we send it multiple nodes at the same time, there is probable chance for split votes.